sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -330,56 +391,38 @@ object SpillableHostColumnarBatch {
    * @param batch         the batch to make spillable
    * @param priority      the initial spill priority of this batch
    */
-  def apply(
-      batch: ColumnarBatch,
-      priority: Long,
-      catalog: RapidsBufferCatalog = RapidsBufferCatalog.singleton): SpillableColumnarBatch = {
+  def apply(batch: ColumnarBatch, priority: Long): SpillableColumnarBatch = {
     val numRows = batch.numRows()
     if (batch.numCols() <= 0) {
       // We consumed it
       batch.close()
       new JustRowsHostColumnarBatch(numRows)
     } else {
       val types = RapidsHostColumnVector.extractColumns(batch).map(_.dataType())
-      val handle = addHostBatch(batch, priority, catalog)
-      new SpillableHostColumnarBatchImpl(
-        handle,
-        numRows,
-        types,
-        catalog)
-    }
-  }
-
-  private[this] def addHostBatch(
-      batch: ColumnarBatch,
-      initialSpillPriority: Long,
-      catalog: RapidsBufferCatalog): RapidsBufferHandle = {
-    withResource(batch) { batch =>
-      catalog.addBatch(batch, initialSpillPriority)
+      val handle = SpillableHostColumnarBatchHandle(batch)
```

Comment:
Why no stream sync if it is needed for the other APIs?

Reply:
this is a host batch, so we didn't need a device sync on this :)